src/narrowphase/narrowphase.cpp

```diff
@@ -1492,14 +1503,19 @@ bool boxBoxIntersect(const Box& s1, const Transform3f& tf1,
   if(contact_points)
   {
     Vec3f contact_point;
-    for(size_t i = 0; i < contacts.size(); ++i)
-    {
-      contact_point += contacts[i].point;
+    if (contacts.size()>0) {
+      std::sort(contacts.begin(), contacts.end(), compareContactPoints);
+      *contact_points = contacts[0].point;
+      if(penetration_depth_) *penetration_depth_ = -contacts[0].depth;
     }
+    // for(size_t i = 0; i < contacts.size(); ++i)
+    // {
+    //   contact_point += contacts[i].point;
+    // }
 
-    contact_point = contact_point / (FCL_REAL)contacts.size();
+    // contact_point = contact_point / (FCL_REAL)contacts.size();
 
-    *contact_points = contact_point;
+    // *contact_points = contact_point;
```

Comment:
Sure, removed by 366a915. I guess the original author (@EdsterG) would think so as well.